core/integration/container_test.go

```diff
@@ -632,32 +632,77 @@ func TestContainerExecWithEntrypoint(t *testing.T) {
 	c, ctx := connect(t)
 
 	base := c.Container().From(alpineImage)
-	before, err := base.Entrypoint(ctx)
-	require.NoError(t, err)
-	require.Empty(t, before)
-
 	withEntry := base.WithEntrypoint([]string{"sh"})
-	after, err := withEntry.Entrypoint(ctx)
-	require.NoError(t, err)
-	require.Equal(t, []string{"sh"}, after)
 
-	used, err := withEntry.WithExec([]string{"-c", "echo $HOME"}).Stdout(ctx)
-	require.NoError(t, err)
-	require.Equal(t, "/root\n", used)
+	t.Run("before", func(t *testing.T) {
+		t.Parallel()
+		before, err := base.Entrypoint(ctx)
+		require.NoError(t, err)
+		require.Empty(t, before)
+	})
+
+	t.Run("after", func(t *testing.T) {
+		t.Parallel()
+		after, err := withEntry.Entrypoint(ctx)
+		require.NoError(t, err)
+		require.Equal(t, []string{"sh"}, after)
+	})
 
-	_, err = withEntry.WithExec([]string{"sh", "-c", "echo $HOME"}).Sync(ctx)
-	require.Error(t, err) // 'sh sh -c echo $HOME' is not valid
+	t.Run("used", func(t *testing.T) {
+		t.Parallel()
+		used, err := withEntry.WithExec([]string{"-c", "echo $HOME"}).Stdout(ctx)
+		require.NoError(t, err)
+		require.Equal(t, "/root\n", used)
+	})
 
-	skipped, err := withEntry.WithExec([]string{"sh", "-c", "echo $HOME"}, dagger.ContainerWithExecOpts{
-		SkipEntrypoint: true,
-	}).Stdout(ctx)
-	require.NoError(t, err)
-	require.Equal(t, "/root\n", skipped)
+	t.Run("prepended to exec", func(t *testing.T) {
+		t.Parallel()
+		_, err := withEntry.WithExec([]string{"sh", "-c", "echo $HOME"}).Sync(ctx)
+		require.Error(t, err) // 'sh sh -c echo $HOME' is not valid
```

Comment:
nit: I know the existing tests often don't follow this, but in general I've been trying to always assert on a specific error message being present rather than just checking error is non-nil. Otherwise it's just too easy for future modifications to result in something error'ing for unexpected reasons but then go unnoticed because it's still technically an error.

Not a big deal at all, but if possible `require.ErrorContains` would be preferable to use here.

Reply:
Yep, agree. I'll make the change 🙂

Reply:
Btw, that was just refactored into separate `t.Run`:

https://github.com/dagger/dagger/blob/1502402c4c028f15165a14ea8f05260057c8141e/core/integration/container_test.go#L629-L661

The new test cases in this test function are "unset default args" and "kept default args":

https://github.com/dagger/dagger/blob/2c85160dfe6e3570cc94b138a3a27409dd4058e9/core/integration/container_test.go#L673-L697

Reply:
Ah okay, I just saw it and commented 😅 Appreciate the improvement either way of course ❤️

Reply:
Updated 🙂